zephyr: FIH brakes CONFIG_SINGLE_APPLICATION_SLOT

build is broken while CONFIG_SINGLE_APPLICATION_SLOT=y
``` console
west build -d build/mcuboot -b nrf52840dk_nrf52840 ./bootloader/mcuboot/boot/zephyr/ -t menuconfig
/* set CONFIG_SINGLE_APPLICATION_SLOT=y */
west build -d build/mcuboot -b nrf52840dk_nrf52840 ./bootloader/mcuboot/boot/zephyr/
...
...
...
eabi/bin/ld: app/libapp.a(single_loader.c.obj): in function `fih_delay_init':
/home/anpu/codedev/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/fault_injection_hardening.h:174: multiple definition of `fih_delay_init'; app/libapp.a(main.c.obj):/home/anpu/codedev/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/fault_injection_hardening.h:174: first defined here
collect2: error: ld returned 1 exit status
```

--- a/boot/bootutil/src/fault_injection_hardening_delay_rng_mbedtls.c
+++ b/boot/bootutil/src/fault_injection_hardening_delay_rng_mbedtls.c
@@ -8,7 +8,6 @@
 
 #ifdef FIH_ENABLE_DELAY
 
-#include "bootutil/fault_injection_hardening_delay_rng.h"
 #include "mcuboot-mbedtls-cfg.h"
 #include "mbedtls/ctr_drbg.h"
 #include "mbedtls/entropy.h"

--- a/boot/cypress/MCUBootApp/main.c
+++ b/boot/cypress/MCUBootApp/main.c
@@ -37,7 +37,6 @@
 #include "bootutil/bootutil_log.h"
 
 #include "bootutil/fault_injection_hardening.h"
-#include "bootutil/fault_injection_hardening_delay_rng.h"
 
 /* Define pins for UART debug output */
 #define CYBSP_UART_ENABLED 1U

--- a/boot/mynewt/src/main.c
+++ b/boot/mynewt/src/main.c
@@ -46,7 +46,6 @@
 #include "bootutil/bootutil.h"
 #include "bootutil/bootutil_log.h"
 #include "bootutil/fault_injection_hardening.h"
-#include "bootutil/fault_injection_hardening_delay_rng.h"
 
 #if MYNEWT_VAL(BOOT_CUSTOM_START)
 void boot_custom_start(uintptr_t flash_base, struct boot_rsp *rsp);

--- a/boot/zephyr/main.c
+++ b/boot/zephyr/main.c
@@ -31,7 +31,6 @@
 #include "bootutil/image.h"
 #include "bootutil/bootutil.h"
 #include "bootutil/fault_injection_hardening.h"
-#include "bootutil/fault_injection_hardening_delay_rng.h"
 #include "flash_map_backend/flash_map_backend.h"
 
 #ifdef CONFIG_MCUBOOT_SERIAL

--- a/boot/zephyr/single_loader.c
+++ b/boot/zephyr/single_loader.c
@@ -10,7 +10,6 @@
 #include "bootutil_priv.h"
 #include "bootutil/bootutil_log.h"
 #include "bootutil/fault_injection_hardening.h"
-#include "bootutil/fault_injection_hardening_delay_rng.h"
 
 #include "mcuboot_config/mcuboot_config.h"
 

